### DISCUSSION TITLE
oscilloscope script

### DISCUSSION THREAD

**Discussion by gemu2015** [arendst/Tasmota#15767] [Show and tell]
just for fun:

ADS1115 oscilloscope script (just input noise, since no input connected)

4 channels in one graph: [ads1115_async_streaming.txt](https://github.com/arendst/Tasmota/files/8850893/ads1115_async_streaming.txt)
4 channels in separate graphs: [ads1115_async_streaming2.txt](https://github.com/arendst/Tasmota/files/8858364/ads1115_async_streaming2.txt)

https://user-images.githubusercontent.com/11647075/172317418-c6768368-c373-4076-9879-c86280a6e79a.mov

**Comment by barbudor** [comment]
Awesome !
The power of script 👍

**Comment by jpdd521** [comment]
有没有人把视频转发一下，被墙了看不到视频。

**Comment by LucidEye** [comment]
This looks awesome!  
I am trying to use an ADS1115 to monitor the voltage levels of a LiPo battery that is powering the ESP32 through an Adafruit PowerBoost 1000 Charger board.
Essentially I want to do some testing running the ESP off some repurposed LiPo batteries I have but also make a little LiPo dis-charger for bringing my batteries down to proper levels for long-term shelf storage.  I'm planning on using a strip of 8 WS2812 LEDs as the discharge load.
Since Tasmota supports controlling these LEDs with on-screen slider controls, I can adjust the load, and thus the discharge rate, by dimming or brightening the LEDs!   
I was also thinking about using a Rule to send a cutoff signal to the PowerBoost board or maybe trip a latching relay to completely cut off the battery and shut down the ESP32 as well so the battery doesn't discharge any further past the desired storage voltage.  
I had originally planned to use MQTT to send the battery data to a Raspberry Pi running my home automation software so I could graph the discharge rate, but with this script it looks like I can graph and visually monitor the progress without having to send the data to external graphing software!
Can this graph be "slowed down" to a slower sampling rate... like longer intervals between the readings on the graph... was thinking about 60 seconds between graph points? 
Also... Does this script go in the Berry console, or is this the "other" script where I need to enable the script function by editing "user_config_override.h" and recompile?
Thank you!

**Reply by gemu2015** [reply]
you can adjust the update interval as you like.
you need scripting (not berry scripting)
be aware that the Graf always restarts when you reload the page, there is currently not history.

**Reply by LucidEye** [reply]
Thank you for the prompt reply!
Sounds good.  No history on the graph is fine... I would probably just leave the web interface tab open to monitor progress... I usually don't leave LiPo batteries totally unattended when charging/dis-charging anyway.  
One other question, is there some way to move the decimal in the analog channel readout so that it shows the actual correct voltage... like 3.340 V instead of 3340.00 V?
I'm also wondering how difficult it would be to run this on a cheap yellow display ESP (CYD)... then it would be a truly stand-alone device and wouldn't even need to connect to the WiFi if I could display the graph locally on the CYD screen! For that matter I could also use the ESP's internal ADC instead of the external ADS1115.
Can't wait to try this!  
Thanks again  :-)

**Reply by gemu2015** [reply]
yes you can scale the ADS or internal ADCs as you like

you may also use a board with display to show your graphs and values.

in this case you should better choose an ESP32 and switch to Berry with LVGL
